app/controllers/application_controller.rb

```diff
@@ -98,4 +95,18 @@ def store_interaction
                                    request.referrer,
                                    study_participant)
   end
+
+  def locale_param
+    return unless params[:locale].in?(available_locales)
+    params[:locale]
+  end
+
+  def cookie_param
```

Comment:
I think this name `cookie_param` is a not straight forward and confusing, because the method does return the locale and not any cookie param (as I would expect). I would suggest  renaming it to `cookie_locale_param`or `locale_cookie_param`.

Reply:
Good point.